### DISCUSSION TITLE
Undercurl not working

### DISCUSSION THREAD

**Discussion by Froxcey** [neovim/neovim#25213] [Q&A] [answered]
I know NVim falls back to underline when undercurl (and other similar stylings) are not available. My NVim keeps falling back to underline despite iTerm2 supporting undercurl. Here are some screenshots:

- iTerm2 supports undercurl:
<img width="351" src="https://github.com/neovim/neovim/assets/51555391/fb2b6b13-7d70-429a-afec-6d5c818cdd65">

- Working when NVim is in Kitty
<img width="445" src="https://github.com/neovim/neovim/assets/51555391/806411ed-8f23-4933-a760-5e76fa89a6cb">

- Working when NVim is in Neovide
<img width="529" alt="Screenshot 2023-09-17 at 6 57 23 PM" src="https://github.com/neovim/neovim/assets/51555391/97c9e8fd-355c-4309-9b73-dea76851d09e">

- Not working when NVim is in iTerm2
<img width="503" alt="Screenshot 2023-09-17 at 5 13 40 PM" src="https://github.com/neovim/neovim/assets/51555391/70e9a41b-fab6-47df-b455-e690662c465c">

System info:
MacOS 13.4.1 (aarch64)
NVim v0.9.2 (termguicolors is enabled)
iTerm 3.4.20 (yes, that's the latest)

Is there some way to force NVim into thinking the terminal supports undercurl (which it actually does)?

<img src="https://github.com/Avdan-OS/Compositor/assets/51555391/d0379882-f2dc-42e1-962f-b3f122db656f" alt="vibe" width="60"/>

**Comment by clason** [comment, marked as answer]
<img width="342" alt="Screenshot 2023-09-17 at 13 58 59" src="https://github.com/neovim/neovim/assets/2361214/81600d81-1d02-4efc-b1fd-18696b3e3a7c">

Works for me on iTerm 3.5.0beta11. (I hate to do this, but avoid the last release; they're outdated in a variety of relevant ways.)

**Reply by Froxcey** [reply]
Yes thank you, it works in 3.5.0beta11
<img width="503" alt="Screenshot 2023-09-17 at 9 16 45 PM" src="https://github.com/neovim/neovim/assets/51555391/62aeb862-8617-4fe6-9182-bd87b2c4161f">

For anyone with the same issue, you can enable beta channel and install the update. That will fix the issue for now, and you can switch back when the stable channel gets the fix.
<img width="897" alt="Screenshot 2023-09-17 at 9 17 58 PM" src="https://github.com/neovim/neovim/assets/51555391/5d90f67c-68ef-42a3-bb73-16428273d1d9">
